### DISCUSSION TITLE
[help]  Loading stantargets objects from outside the targets workflow

### DISCUSSION THREAD

**Discussion by bmaitner** [wlandau/stantargets#60] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

Is it possible to load a stantargets object from outside of a targets directory? For example, with other targets files in the _targets/objects folder , I can load these using readRDS().  However, whenever I try this with files generated by stantargets (e.g., summary or draws objects), I receive an error stating "unknown input format".

``` r
# Make a temp directory
  temp <- tempdir()

# Link to a targets object created by targets  
  stantargets_object <- "https://github.com/AdamWilsonLab/emma_model/raw/targets-runs/_targets/objects/model_output_data_e4183434"
  
# Link to a targets object created by stantargets  
targets_object <- "https://github.com/AdamWilsonLab/emma_model/raw/targets-runs/_targets/objects/dyndata_training"

# Download files to temp dir
  download.file(url = stantargets_object,
                destfile = file.path(temp,basename(stantargets_object)))
  
  download.file(url = targets_object,
                destfile = file.path(temp,basename(targets_object)))

# Attempt to load files:  
  readRDS(file.path(temp, basename(stantargets_object))) #"Unknown input format"
#> Error in readRDS(file.path(temp, basename(stantargets_object))): unknown input format

  readRDS(file.path(temp, basename(targets_object)))|>
    head() # Works fine
#>    cellID       date most_recent_burn_date ndvi time_since_fire age
#> 1 2895983 2000-12-01                     0   NA              NA  NA
#> 2 2895985 2000-12-01                     0   NA              NA  NA
#> 3 2895986 2000-12-01                     0   NA              NA  NA
#> 4 2898019 2000-12-01                     0   NA              NA  NA
#> 5 2898020 2000-12-01                     0   NA              NA  NA
#> 6 2900051 2000-12-01                     0   NA              NA  NA
```

<sup>Created on 2023-07-21 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by wlandau** [comment, marked as answer]
My recommendation at https://github.com/ropensci/targets/discussions/1097#discussioncomment-6513356 would help if the files are local, but it looks like you are loading it from a GitHub repo. Also, `stantargets` uses `format = "qs"` and `format = "fst"`, so you might need `qs::qread()` and/or `fst::read_fst()` for the code in your comment to work.

If you want to store the data remotely and access it locally, you could use built-in cloud storage: https://books.ropensci.org/targets/data.html#cloud-storage

**Reply by bmaitner** [reply]
Ahh, I see.  I was under the impression that everything without an extension in the _targets/objects folder was an RDS file (per https://books.ropensci.org/targets-design/data.html#targets ).  Now that I know it contains the other types, I can put together a little helper function to load objects of unknown type.  Thanks so much for the quick reply and the awesome tools!  :)
